quickwit/quickwit-opentelemetry/src/otlp/traces.rs

```diff
@@ -753,22 +760,28 @@ impl OtlpGrpcTracesService {
     fn parse_spans(
         request: ExportTraceServiceRequest,
         parent_span: RuntimeSpan,
-        index_id: IndexId,
     ) -> tonic::Result<ParsedSpans> {
         let spans = parse_otlp_spans(request)?;
         let num_spans = spans.len() as u64;
         let mut num_parse_errors = 0;
         let mut error_message = String::new();
 
-        let mut doc_batch_builder = DocBatchBuilder::new(index_id).json_writer();
+        let mut doc_batch_builder = DocBatchV2Builder::default().json_writer();
+        let mut doc_uid_generator = DocUidGenerator::default();
         for span in spans {
-            if let Err(error) = doc_batch_builder.ingest_doc(&span.0) {
+            let doc_uid = doc_uid_generator.next_doc_uid();
+            if let Err(error) = doc_batch_builder.add_doc(doc_uid, span.0) {
```

Comment:
idea: we could have a `add_json_doc` function on the `DocBatchV2Builder`.

Reply:
I copied over the `JsonDocBatchBuilder` logic from V1. To use `DocBatchV2Builder` directly, we would either:
- need to first serialize the struct then add thoses bytes to the `BytesMut`, but that means an extra copy
- also use a `Writer<BytesMut>` in `DocBatchV2Builder`, but I assumed it was not 0 cost otherwise it would have been done that way in V1

Reply:
Following the other review comments, I simplified the `JsonDocBatchV2Builder` as much as possible by avoiding the unnecessary forth and back with `DocBatchV2Builder`. I would prefer not to unify the two structures without ensuring that there isn't a performance impact as this is on the critical write path. What do you think?

Reply:
fine for me!